internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -130,6 +150,10 @@ func New(sess *session.Session) CloudFormation {
 		appStackSet: stackset.New(sess),
 		s3Client:    s3.New(sess),
 		region:      aws.StringValue(sess.Config.Region),
+		console:     new(discardFile),
+	}
+	for _, opt := range opts {
+		opt(&client)
 	}
 	return client
```

Comment:
Just a question for my own purposes: why do we use a struct receiver and not a pointer receiver for the cf methods? Do we want to disallow the cfn client from modifying itself?

Reply:
I have no idea why we did it that way! It's a good question, tbh if I were to write it today I'd make it a pointer receiver to slightly save on memory but if I had to guess your answer is correct!